features/home/src/main/java/com/android254/droidconKE2020/home/ui/viewmodel/HomeViewModel.kt

```diff
@@ -37,44 +37,47 @@ class HomeViewModel(
     private val _keynoteSpeaker = MutableLiveData<SpeakerUIModel>()
     val keynoteSpeaker: LiveData<SpeakerUIModel> get() = _keynoteSpeaker
 
-    fun fetchAllSessions() {
-        viewModelScope.launch {
-            when (val value = sessionRepository.fetchAllSessions()) {
-                is Data.Success -> _sessions.postValue(value.data)
-                is Data.Error -> showToast.postValue(value.exception.toString())
+    fun loadData() = viewModelScope.launch {
+        fetchAllSessions()
+        fetchSponsors()
+        fetchSpeakers()
+        fetchOrganizers()
+    }
```

Comment:
On this my suggestion was we do the calls in parallel to reduce the time take eg

async {fetchAllSessions()}
async {fetchSponsors()}

With this approach, these network calls would  take let's say 3 secs for all of them but the launch one would be 3 secs* no of requests